src/backend.py

```diff
@@ -0,0 +1,726 @@
+from  ujson import dumps as json_dumps 
+from hashlib import sha256 as hashlib_sha256
+from binascii import hexlify
+from time import time, sleep
+from gc import collect as gc_collect, threshold as gc_threshold
+from uctypes import bytearray_at
+from machine import RTC
+
+from phew import server
+
+from SPI_DataStore import (
+    writeIP, 
+    read_record as ds_read_record, 
+    write_record as ds_write_record, 
+    memory_map as ds_memory_map
+)
+from Utilities.random_bytes import random_hex
+from Utilities.ls import ls
+from Memory_Main import save_ram, blank_ram
+from Shadow_Ram_Definitions import SRAM_DATA_BASE, SRAM_DATA_LENGTH, SRAM_COUNT_BASE
+from GameStatus import report as game_status_report
+import Pico_Led
+import faults
+
+#
+# Constants
+#
+rtc = RTC()
+ram_access = bytearray_at(SRAM_DATA_BASE,SRAM_DATA_LENGTH)
+WIFI_MAX_ATTEMPTS = 12
+AP_NAME = "Warped Pinball"
+# Authentication variables
+current_challenge = None
+challenge_timestamp = None
+CHALLENGE_EXPIRATION_SECONDS = 60
+
+
+#
+# Standardized Route Functions
+#
+def route_wrapper(func):
+    def wrapped_route(request):
+        gc_collect()
+        try:
+            response = func(request)
+
+            if response is None:
+                return "ok", 200
+            return response
+
+        except Exception as e:
+            msg = f"Error in {func.__name__}: {e}"
+            print(msg)
+            print(request)
+            print(request.query)
+            return msg, 500
+
+        finally:
+            gc_collect()
+    return wrapped_route
```

Comment:
This standardizes garbage collection and error handling, it's why very few of the routes handle their own errors.